core/src/test/scala/com/salesforce/op/stages/impl/feature/TextMapLenEstimatorTest.scala

```diff
@@ -0,0 +1,109 @@
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.salesforce.op.stages.impl.feature
+
+import com.salesforce.op.features.types.{TextMap, _}
+import com.salesforce.op.test.TestOpVectorColumnType.DescColWithGroup
+import com.salesforce.op.test.{TestFeatureBuilder, TestOpVectorMetadataBuilder, TestSparkContext}
+import com.salesforce.op.utils.spark.{OpVectorColumnMetadata, OpVectorMetadata}
+import com.salesforce.op.utils.spark.RichDataset._
+import org.apache.spark.ml.linalg.Vectors
+import org.junit.runner.RunWith
+import org.scalatest.FlatSpec
+import org.scalatest.junit.JUnitRunner
+
+
+@RunWith(classOf[JUnitRunner])
+class TextMapLenEstimatorTest extends FlatSpec with TestSparkContext with AttributeAsserts {
+
+  val (ds, f1) = TestFeatureBuilder(
+    Seq[TextMap](
+      TextMap(Map("k1" -> "A giraffe drinks by the watering hole", "k2" -> "Cheese", "k3" -> "Hello", "k4" -> "Bye")),
+      // scalastyle:off
+      TextMap(Map("k2" -> "French Fries", "k4" -> "\uA7BC\u10C8\u2829\u29BA\u23E1")),
+      // scalastyle:on
+      TextMap(Map("k3" -> "Hip-hop Pottamus"))
+    )
+  )
+
+  Spec[TextMapLenEstimator[_]] should "take an array of features as input and return a single vector feature" in {
+    val vectorizer = new TextMapLenEstimator[TextMap]().setInput(f1)
+    val vector = vectorizer.getOutput()
+
+    vector.name shouldBe vectorizer.getOutputFeatureName
+    vector.typeName shouldBe FeatureType.typeName[OPVector]
+    vector.isResponse shouldBe false
+  }
+
+  it should "transform the data correctly" in {
+    val vectorizer = new TextMapLenEstimator[TextMap]().setInput(f1)
+    val transformed = vectorizer.fit(ds).transform(ds)
+    val vector = vectorizer.getOutput()
+
+    val expected = Array(
+      Array(31.0, 6.0, 5.0, 3.0),
+      Array(0.0, 11.0, 0.0, 0.0),
+      Array(0.0, 0.0, 14.0, 0.0)
+    ).map(Vectors.dense(_).toOPVector)
+
+    val result = transformed.collect(vector)
+    result shouldBe expected
+
+    val field = transformed.schema(vector.name)
+    assertNominal(field, Array.fill(expected.head.value.size)(false), result)
+    val vectorMetadata = vectorizer.getMetadata()
+    OpVectorMetadata(vectorizer.getOutputFeatureName, vectorMetadata) shouldEqual TestOpVectorMetadataBuilder(
+      vectorizer,
+      f1 -> List(
+        DescColWithGroup(name = Option(OpVectorColumnMetadata.TextLenString), groupName = "k1"),
+        DescColWithGroup(name = Option(OpVectorColumnMetadata.TextLenString), groupName = "k2"),
+        DescColWithGroup(name = Option(OpVectorColumnMetadata.TextLenString), groupName = "k3"),
+        DescColWithGroup(name = Option(OpVectorColumnMetadata.TextLenString), groupName = "k4")
+      )
+    )
+  }
+
+  // TODO: Not sure if we should actually allow this option since text will still be tokenized no matter what...
```

Comment:
Do we need this TODO?

Reply:
I removed the ability to set text cleaning to true since we're tokenizing instead